backend/internal/temporalclient/config.go

```diff
@@ -20,14 +20,18 @@ type Config struct {
 	LogLevel              zap.AtomicLevel `koanf:"log_level"`
 
 	// DevServer.ClientOptions is not used.
-	DevServer testsuite.DevServerOptions `koanf:"dev_server"`
+	DevServer       testsuite.DevServerOptions `koanf:"dev_server"`
+	TLSEnabled      bool                       `koanf:"tls_enabled"`
```

Comment:
move this into its own structure, `TLSConfig`? Then reference it here as `TLS TLSConfig `koanf:"tls"`